### PR TITLE
Display CI Information (CI-badge)

### PR DESCRIPTION
**Pull request by max-krichenbauer**
(see Issue https://github.com/ros-infrastructure/rosindex/issues/78 for some background)

This PR adds information on the CI (continuous integration) status of the package displayed.

If the package has no information about it on the doc server, a negative "no CI" badge is displayed:
image

If the package has a `manifest.yaml` with dev job information, then a positive CI badge is displayed which is also a link to that job:
image

If additionally test data is available (`results.yaml`), then the results of recent tests are also displayed as a drop-down menu, where each item in the list is a link to that specific job:
image

### CODE REVIEW

File: _includes/package_body.html
Comment (tfoote) [inline _includes/package_body.html]: This should be pushed down to the repository summary as all of our CI tests actually run on the repository not on the package level.

File: _includes/package_body.html
Comment (tfoote) [inline _includes/package_body.html]: Please move this styling into the css for consistency. And can we enable color changing based on the success status?

**Review by tfoote — CHANGES_REQUESTED** [review]
Thanks for pushing on this. It will be great to have added. 

I'd originally envisioned the badges as doing the usual javascript loading from a remote resource which means that they're always up to date. This does have the benefit that we won't end up hammering the jenkins instance with a ton of little queries for the images. But on the flip side it might be showing out of date information (up to 24 hours) if there's been a recent fix. Could we put a little timestamp notation display of how long ago the data was rendered into the box? Maybe a little bit more subtlly though. But to be clear the current state of the data. This would be especially valuable if say the indexer goes offline for some reason. It's possible that it could be promoted to the top right of the header or somewhere like that since it's true for all the package info.

Screenshot from 2019-11-14 17-29-24

**Comment by dirk-thomas** [comment]
> Could we put a little timestamp notation display of how long ago the data was rendered into the box?

How about making it a tooltip on the badge?

**Comment by max-krichenbauer** [comment]
Thank you for your feedback. I implemented it as good as I could.

(1)  Styling now done via CSS (in `css/prettify.css`, I hope that's the correct place)

(2) Coloring based on build / test result:
image

(3) Timestamp of information now shown in the tooltip (and in the history drop-down as well if available)
image

File: _includes/package_body.html
Comment (max-krichenbauer) [reply _includes/package_body.html]: Done. Please see comment below commit.

File: _includes/package_body.html
Comment (max-krichenbauer) [reply _includes/package_body.html]: Oh, I see. In this case it was wrong to search for the package name and not the repo name.

I changed that and moved it down to the other box, but now I'm not sure how to deal with packages that don't have a `devel_jobs/.../results.yaml` on the doc server - which are quite a lot.
The wiki code just seems to give them a CI badge anyways.
I thought this was a bug / missing feature and made it look for the `api/.../manifest.yaml`.
What is the expected behavior for packages that are on the build farm and have build success or failure recorded in their `manifest.yaml`, but no `results.yaml` for their repo?

Right now, the flow is:
(1) check for `manifest.yaml` (package-base)
If none is found, show "No CI" bagde.
(2) check for `results.yaml` (repository based)
If none is found, show simple CI badge with no additional information.
(3) parse `results.yaml` for additional information and display extended badge with test results if available.

**Comment by gbiggs** [comment]
For the case where there is CI but no results, what will be displayed? Just saying "there is CI" gives no indication that results are not available, which in itself could be a problem to the viewer.

**Comment by max-krichenbauer** [comment]
@gbiggs 
You can see the screenshot in the original message (the second screenshot). Basically, hovering with the mouse will show the tooltip "No test statistics available for this package".

**Review by tfoote — APPROVED** [review]
Thanks for the improvements. This looks good. I'd merge it but I'm going to hold off until we've resolved #200

**Comment by max-krichenbauer** [comment]
Sorry for being pushy, but is there anything that still should be changed about this PR or something else that prohibits merging it? #200 seems to have been solved.

**Comment by tfoote** [comment]
No problem. Thanks for the followup. It's hard to catch all the followups after having to wait for the build results after that merge.

**Comment by itualami** [comment]
Thanks to both @max-krichenbauer @tfoote of you for this effort. I just noticed this has been merged. Great effort!

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/next-ros-quality-assurance-working-group-february-2020/12571/1

**Pull request merged**
